src/server/acl/validator.cc

```diff
@@ -31,12 +35,51 @@ namespace dfly::acl {
 
 [[nodiscard]] bool IsUserAllowedToInvokeCommandGeneric(uint32_t acl_cat,
                                                        const std::vector<uint64_t>& acl_commands,
+                                                       const AclKeys& keys, CmdArgList tail_args,
                                                        const facade::CommandId& id) {
   const auto cat_credentials = id.acl_categories();
   const size_t index = id.GetFamily();
   const uint64_t command_mask = id.GetBitIndex();
   DCHECK_LT(index, acl_commands.size());
-  return (acl_cat & cat_credentials) != 0 || (acl_commands[index] & command_mask) != 0;
+
+  const bool is_read_command = id.opt_mask() & CO::CommandOpt::READONLY;
+  const bool is_write_command = id.opt_mask() & CO::CommandOpt::WRITE;
+  const size_t step = (id.opt_mask() & CO::CommandOpt::INTERLEAVED_KEYS) ? 2 : 1;
+  const bool var_keys = (id.opt_mask() & CO::CommandOpt::VARIADIC_KEYS);
+
+  auto match = [](const auto& pattern, const auto& target) {
+    return stringmatchlen(pattern.data(), pattern.size(), target.data(), target.size(), 0);
```

Comment:
wow wtf this is a built-in redis function

Reply:
yes!